src/seals/base_envs.py

```diff
@@ -278,43 +266,42 @@ def __init__(
         self.horizon = horizon
         self.initial_state_dist = initial_state_dist
 
-        super().__init__(
-            state_space=self._construct_state_space(),
-            action_space=self._construct_action_space(),
-            observation_space=self._construct_observation_space(),
-        )
-
-    def _construct_state_space(self) -> gym.Space:
-        return spaces.Discrete(self.state_dim)
-
-    def _construct_action_space(self) -> gym.Space:
-        return spaces.Discrete(self.action_dim)
-
-    @abc.abstractmethod
-    def _construct_observation_space(self) -> gym.Space:
-        pass  # pragma: no cover
+        self.state_space = spaces.Discrete(self.state_dim)
```

Comment:
Ah I see why we were introducing a setter now. This does have the benefit we can call the superclass constructor first rather than last (which is a weird thing to do OO wise). But the object isn't really in a consistent state until we run these setter calls (since it's missing mandatory observations_space and action_space attributes). And there's a risk with this design that we forgot to initialize observation_space/action_space. WDYT?